modules/ding_ddbasic/ding_ddbasic.module

```diff
@@ -79,7 +79,35 @@ function ding_ddbasic_form_alter(&$form, &$form_state, $form_id) {
           ),
           '#size' => '1',
           '#default_value' => $default_value,
-        ) + $form['color']['palette']['text_link'];
+        ) + (isset($form['color']['palette']['text_link']) ? $form['color']['palette']['text_link'] : array());
+
+        // Limit the choices on text link colors on dark background.
+        $default_value = '#ffffff';
+        if (isset($form['color']['palette']['text_link_on_dark']['#default_value'])) {
+          switch ($form['color']['palette']['text_link_on_dark']['#default_value']) {
+            case $form['color']['palette']['primary']['#default_value']:
+              $default_value = 'primary';
+              break;
+
+            case $form['color']['palette']['secondary']['#default_value']:
+              $default_value = 'secondary';
+              break;
+          }
+        }
+
+        $form['color']['palette']['text_link_on_dark'] = array(
+          '#description' => t('Choose a color that is legible on a dark background<br />Is used for text-links on dark background'),
+          '#type' => 'select',
+          '#options' => array(
+            // The primary and secondary value is handled in the theme_settings
+            // submit callback, in the ddbasic theme.
+            'primary' => t('Primary'),
+            'secondary' => t('Secondary'),
+            DING_DDBASIC_COLOR_WHITE => t('White'),
+          ),
+          '#size' => '1',
+          '#default_value' => $default_value,
+        ) + (isset($form['color']['palette']['text_link_on_dark']) ? $form['color']['palette']['text_link_on_dark'] : array());
```

Comment:
Why are these changes included in this PR? They do not seem to be related to the changes at hand.

Reply:
Yes they are.. as php 5.6 ignores these but PHP 7.0 throws an error that stops PHP execution. So the theme settings page is not accessible when using php 7